src/external/monaco.flux.messages.ts

```diff
@@ -167,13 +167,16 @@ export const parseResponse = (response: ServerResponse): LSPResponse => {
     throw new Error(error)
   }
 
-  const split = (message || '').split('\r\n')
+  if (message) {
+    const split = (message || '').split('\r\n')
 
-  try {
-    return JSON.parse(split.slice(2).join('\n'))
-  } catch (e) {
-    throw new Error('failed to parse LSP response')
+    try {
+      return JSON.parse(split.slice(2).join('\n'))
+    } catch (e) {
+      throw new Error('failed to parse LSP response', e)
+    }
   }
+  return undefined
```

Comment:
This is probably a question for the UI team specifically, but `return undefined` is equivalent to no return at all. Not sure if we want to be explicit here or not.

Reply:
fair point. I thought typescript would complain without the return, but looks like it doesn't care so I'll remove this